### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by Sspadea**
## Summary

- Endpoint post /users was created and is working correctly.

## Trello Card

https://trello.com/c/pY0tFweL/35-sign-up

### CODE REVIEW

File: app/middlewares/checkValidations.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const { validationResult } = require('express-validator');
+const { validationError } = require('../errors');
+
+const checkValidations = (req, res, next) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) throw validationError(errors.array());
```
Comment (DiegoWolox) [inline app/middlewares/checkValidations.js]: to pass errors to express is through the function next
`return next(validationError(errors.array()))`

with the return it does not continue executing the code (execute the next twice)

File: app/services/user.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const bcrypt = require('bcryptjs');
+const User = require('../models').Users;
+
+const encryptPassword = async password => {
+  const salt = await bcrypt.genSalt(10);
+  const hash = await bcrypt.hash(password, salt);
```
Comment (DiegoWolox) [inline app/services/user.js]: you can also return this directly
`return bcrypt.hash(password, salt);`

File: app/services/user.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+exports.createUser = async data => {
+  const hash = await encryptPassword(data.password);
+  const newUser = await User.create({
+    firstName: data.first_name,
+    lastName: data.last_name,
+    email: data.email.trim().toLowerCase(),
+    password: hash
+  });
+  return newUser;
```
Comment (DiegoWolox) [inline app/services/user.js]: same here, you can directly return the promise

File: app/validations/user.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { check } = require('express-validator');
+const errorMessages = require('../middlewares/errors');
+
+exports.validateUser = [
```
Comment (DiegoWolox) [inline app/validations/user.js:4]: nice!!

File: app/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const { createUser } = require('../services/user');
+const asyncWrapper = require('../utils/asycWrapper');
+
+exports.newUser = asyncWrapper(async (req, res) => {
+  const user = await createUser(req.body);
+  return res.status(200).json({ first_name: user.firstName });
```
Comment (DiegoWolox) [inline app/controllers/users.js]: when you create an entity (user) the response is 201 (Created)

Comment (wgermancutraro) [reply app/controllers/users.js]: You can create a constant file that returns all the status codes. { CREATED: 201 } and use that

**Review by DiegoWolox — DISMISSED** [review]

**Review by DiegoWolox — APPROVED** [review]

**Review by wgermancutraro — APPROVED** [review]

**Pull request merged**